src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -133,6 +138,24 @@ export default Vue.extend({
     'click-outside': clickOutside,
   },
   methods: {
+    // filter to check if a course fulfills a requirements checker
+    canFulfillChecker(
+      { userRequirementsMap, toggleableRequirementChoices }: VuexStoreState,
+      requirementId: string,
+      course: FirestoreSemesterCourse
+    ) {
+      const requirement = userRequirementsMap[requirementId];
+      // If we cannot find the relevant requirement, then default to true to be permissive.
+      if (requirement == null) return () => true;
+      const requirementSpec = getMatchedRequirementFulfillmentSpecification(
+        requirement,
+        toggleableRequirementChoices
+      );
+      // If a requirement is truly self-check, then all courses can be used.
+      if (requirementSpec == null) return () => true;
+      const eligibleCourseIds = new Set(requirementSpec.eligibleCourses.flat());
+      return eligibleCourseIds.has(course.crseId);
+    },
```

Comment:
This could be good to put in `requirement-frontend-utils.ts` in case we want to reuse this functionality in other places later on.

Reply:
I agree, especially because I based this code heavily off of @SamChou19815 's code in #383 so there is already some code re-use. However, migrating it to `requirement-frontend-utils.ts` results in the following error when trying to import `import { VuexStoreState } from '@/store';`

`Dependency cycle via ./requirements/requirement-frontend-computation:5eslintimport/no-cycle`

Looks like this is because `store.ts` also imports `requirement-frontend-utils.ts.` Is there somewhere else I should put this code, or should I leave it like it is for now?

Reply:
Ah right we want to avoid cyclical imports. I suspected this could be a problem....what if instead of  
`{ userRequirementsMap, toggleableRequirementChoices }: VuexStoreState` your function would have the params `userRequirementsMap: Readonly<Record<string, RequirementWithIDSourceType>>` and `toggleableRequirementChoices: AppToggleableRequirementChoices`? Then you wouldn't have to import the store. Does that make sense?

See:

https://github.com/cornell-dti/course-plan/blob/b5f1ae5fe25fc15f4c6d5aefe6c03dfca0da8911/src/store.ts#L46-L48